experimental/Schemes/Resolution_structure.jl

```diff
@@ -736,3 +731,58 @@ end
 
 is_graded(R::Ring) = false
 
+########################################################################
+# Implement the interface specified in 
+# experimental/Schemes/BlowupMorphism.jl
+########################################################################
+
+function exceptional_divisor(phi::BlowUpSequence)
+  #TODO: Check whether the full resolution has already been computed?
+  if !isdefined(phi, :exceptional_divisor)
+    phi.exceptional_divisor = sum(phi.ex_div; init=CartierDivisor(domain(phi), ZZ))
+  end
+  return phi.exceptional_divisor
+end
+
+function exceptional_locus(phi::BlowUpSequence)
+  if !isdefined(phi, :exceptional_locus)
+    phi.exceptional_locus = weil_divisor(exceptional_divisor(phi))
+  end
+  return phi.exceptional_locus
+end
```

Comment:
This is the point where we need to discuss. For non-embedded desingularization this is what we need. But in the embedded case and for principalization of Ideal Sheaves we do weak transforms and need to store multiplicities along with the reduced structure.
I could envision that we additionally add a function which returns a Weil divisor with the appropriate multiplicities -- from the tradition of the resolution of singularities community, this would be called `monomial_part`. But this name is absolutely impossible in the computeralgebra context, because it is taken otherwise. So I would envision `exceptional_factors_with_multiplicities`.

Reply:
I am not sure that I get 100% what you mean here. Let's discuss this in person at some point. 

Until then: Does my implementation give wrong results? If not, it sounds to me as if we could leave it as is for the moment and later add further functions or modify the implementation.